### PR TITLE
feat: update to @ui5/webcomponents 0.31.15 & downport `AnalyticalTable`

### PR DESCRIPTION
**Pull request by Lukas742**
The downport of the `AnalyticalTable` mainly includes the keyboard navigation feature (#1864), but since we couldn't update just that also all other changes up until [v0.17.4](https://github.com/SAP/ui5-webcomponents-react/releases/tag/v0.17.4) are included as well.

The last downported version of the table was the one of [v0.14.2](https://github.com/SAP/ui5-webcomponents-react/releases/tag/v0.14.2).

### CODE REVIEW

File: packages/main/src/webComponents/SegmentedButton/SegmentedButton.stories.mdx [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
-import { ToggleButton } from '@ui5/webcomponents-react/lib/ToggleButton';
+import { SegmentedButtonItem } from '@ui5/webcomponents-react/lib/SegmentedButtonItem';
 import { createSelectArgTypes } from '@shared/stories/createSelectArgTypes';
 import { DocsHeader } from '@shared/stories/DocsHeader';
 import { DocsCommonProps } from '@shared/stories/DocsCommonProps';
 
 <Meta
   title="Inputs / SegmentedButton"
   component={SegmentedButton}
+  subcomponents={SegmentedButtonItem}
```
[the remark this answers is not in the document]
Comment (Lukas742) [reply packages/main/src/webComponents/SegmentedButton/SegmentedButton.stories.mdx]: Yup, done :)

File: packages/main/src/webComponents/SegmentedButtonItem/index.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+   */
+  iconEnd?: unknown;
+  /**
+   * **Note:** The property is inherited and not supported. If set, it won't take any effect.
+   */
+  iconSize?: unknown;
+  /**
+   * **Note:** The property is inherited and not supported. If set, it won't take any effect.
+   */
+  submits?: unknown;
```
[the remark this answers is not in the document]
Comment (Lukas742) [reply packages/main/src/webComponents/SegmentedButtonItem/index.tsx]: Done

File: packages/main/src/webComponents/SegmentedButtonItem/index.tsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+/**
+ * Users can use the `SegmentedButtonItem` as part of a `SegmentedButton`.
+ *
+ * Clicking or tapping on a `SegmentedButtonItem` changes its state to `pressed`. The item returns to its initial state when the user clicks or taps on it again. By applying additional custom CSS-styling classes, apps can give a different style to any `SegmentedButtonItem`.
+ *
+ * <a href="https://sap.github.io/ui5-webcomponents/playground/components/SegmentedButtonItem" target="_blank">UI5 Web Components Playground</a>
+ */
+const SegmentedButtonItem: FC<SegmentedButtonItemPropTypes> = withWebComponent<SegmentedButtonItemPropTypes>(
+  'ui5-segmented-button-item',
+  ['design', 'iconEnd', 'iconSize', 'submits', 'icon'],
```
[the remark this answers is not in the document]
Comment (Lukas742) [reply packages/main/src/webComponents/SegmentedButtonItem/index.tsx]: Done

File: packages/main/src/webComponents/ShellBarItem/index.tsx [hunk 4]
```diff
@@ -21,8 +21,6 @@ export interface ShellBarItemPropTypes extends WithWebComponentPropTypes {
   text?: string;
   /**
    * Fired, when the item is pressed.
-   *
```
[the remark this answers is not in the document]
Comment (Lukas742) [reply packages/main/src/webComponents/ShellBarItem/index.tsx:24]: Yes, there was an unnecessary empty line. Not 100% sure if I removed it manually or it was removed by the script though :D

**Review by MarcusNotheis — APPROVED** [review]

**Pull request merged**